medusa/helper/externals.py

```diff
@@ -0,0 +1,161 @@
+# coding=utf-8
+# Author: p0psicles
+#
+# This file is part of Medusa.
+#
+# Medusa is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Medusa is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Medusa. If not, see <http://www.gnu.org/licenses/>.
+# pylint:disable=too-many-lines
+"""Externals helper functions."""
+
+import logging
+from traktor import TokenExpiredException, TraktApi, TraktException
+from .. import app
+from ..indexers.indexer_api import indexerApi
+from ..indexers.indexer_config import indexerConfig
+from ..indexers.indexer_exceptions import IndexerShowAllreadyInLibrary, IndexerUnavailable
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_trakt_externals(externals):
+    """Small trakt api wrapper, to request trakt externals using multiple external id's.
+
+    :param externals: Dictionary of key/value pairs with external id's.
+    """
+    def trakt_request(api, url):
+        """Perform the request and handle possible token refresh."""
+        try:
+            result = api.request(url) or []
+            if api.access_token_refreshed:
+                app.TRAKT_ACCESS_TOKEN = api.access_token
+                app.TRAKT_REFRESH_TOKEN = api.refresh_token
+                app.instance.save_config()
+        except (TokenExpiredException, TraktException):
+            app.TRAKT_ACCESS_TOKEN = ''
+            raise
+        return result
+
+    trakt_settings = {'trakt_api_key': app.TRAKT_API_KEY,
+                      'trakt_api_secret': app.TRAKT_API_SECRET,
+                      'trakt_access_token': app.TRAKT_ACCESS_TOKEN,
+                      'trakt_refresh_token': app.TRAKT_REFRESH_TOKEN}
+    trakt_api = TraktApi(app.SSL_VERIFY, app.TRAKT_TIMEOUT, **trakt_settings)
+
+    id_lookup = '/search/{external_key}/{external_value}?type=show'
+    trakt_mapping = {'tvdb_id': 'tvdb', 'imdb_id': 'imdb', 'tmdb_id': 'tmdb', 'trakt_id': 'trakt'}
+    trakt_mapping_rev = {v: k for k, v in trakt_mapping.items()}
+
+    for external_key in externals:
+        if not trakt_mapping.get(external_key) or not externals[external_key]:
```

Comment:
You are right, didn't notice it, the second check can be removed completely then.